src/openrct2/object/ObjectManager.cpp

```diff
@@ -232,6 +233,22 @@ class ObjectManager final : public IObjectManager
         return objects;
     }
 
+    static rct_string_id GetObjectSourceGameString(const rct_object_entry * entry)
+    {
+        uint8 source = (entry->flags & 0xF0) >> 4;
```

Comment:
I could have sworn that this code or something very similar already existed and was used for error message reporting. It's possible it was removed though.